pubtools/_content_gateway/push_staged_cgw.py

```diff
@@ -0,0 +1,185 @@
+from pushsource import CGWPushItem
+from .cgw_client import *
+from .cgw_authentication import CGWBasicAuth
+from .utils import yaml_parser, validate_schema
+import logging
+from attrs import asdict
+import pluggy
+import json
+
+pm = pluggy.PluginManager("pubtools")
+hookspec = pluggy.HookspecMarker("pubtools")
+hookimpl = pluggy.HookimplMarker("pubtools")
+
+LOG = logging.getLogger("pubtools.cgw")
+LOG_FORMAT = "%(asctime)s [%(levelname)-8s] %(message)s"
+logging.basicConfig(level=logging.DEBUG, format=LOG_FORMAT)
+
+
+class PushStagedCGW:
+    def __init__(self, target_name, target_settings):
+        self.auth = CGWBasicAuth(target_settings['username'], target_settings['password'])
+        self.cgw_client = CGWClient(target_settings['server_name'], self.auth)
+        self.product_mapping = {}
+        self.pv_mapping = {}
+        self.push_items = []
+        self.target_name = target_name
+        self.pulp_push_items = {}
+
+    def _create_product_version_mapping(self, product_name, product_code):
+        product_id = self._get_product_id(product_name, product_code)
+        all_versions = self.cgw_client.get_versions(product_id)
+        for data in all_versions:
+            self.pv_mapping[(product_name, product_code, data.get('versionName'))] = (product_id, data.get('id'))
+
+    def _get_product_id(self, product_name, product_code):
+        product_id = self.product_mapping.get((product_name, product_code))
+        return product_id
+
+    def _get_version_id(self, product_name, product_code, version_name):
+        version_id = self.pv_mapping.get((product_name, product_code, version_name)) if self.pv_mapping else None
+        if version_id:
+            _, version_id = version_id
+        return version_id
+
+    def _get_file_id(self, file_item):
```

Comment:
this looks very same as PushCGW._get_file_id, so please create Common class which both classes inherits from